lib/spack/spack/build_systems/cuda.py

```diff
@@ -185,3 +185,14 @@ def cuda_flags(arch_list):
     # Make sure cuda_arch can not be used without +cuda
     for value in cuda_arch_values:
         conflicts('~cuda', when='cuda_arch=' + value)
+
+    def setup_build_environment(self, env):
+        build_type_flags = self.build_type.get_flags('cudaflags')
+        env.set('NVCC_PREPEND_FLAGS', build_type_flags)
+
+    def flag_handler(self, name, flags):
+        if self.build_type:
+            debug_flags = ('cflags', 'cxxflags', 'fflags')
+            if self.build_type.name == "debug" and name in debug_flags:
+                flags.append('-G')
+        return super(CudaPackage, self).flag_handler(name, flags)
```

Comment:
I don't think we need this method now that we have the flags in `setup_build_environment`

Reply:
Oh I meant to delete this, shouldn’t be there!